subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinScriptEvaluator.kt

```diff
@@ -376,7 +379,7 @@ class CompileKotlinScript(
         identityFileInputs: MutableMap<String, CurrentFileCollectionFingerprint>
     ): UnitOfWork.Identity {
         val identityHash = newHasher().let { hasher ->
-            listOf("templateId", "sourceHash", "compilationClassPath", "accessorsClassPath").forEach {
+            listOf("assignmentOverloadEnabled", "templateId", "sourceHash", "compilationClassPath", "accessorsClassPath").forEach {
```

Comment:
🤔 Would be nice to move these strings out into constants to avoid typos.

Reply:
It might be we missed to add it to the identity hash

Reply:
Maybe it would be worth extracting this `listOf()` as a constant close to the declaration of the other constants to avoid missing another property later?

Reply:
Moved the list to constants